### ISSUE TITLE
[EXPO][EAS][iOS]["useFrameworks": "static"] Build fails to resolve from react-native bridging headers

### ISSUE THREAD

**Issue by atanaskanchev** [nrwl/nx#14014]
### Current Behavior

Local eas iOS build fails to resolve `react/bridging/CallbackWrapper.h` and `react/bridging/LongLivedObject.h` from the react-native bridging headers. There's no such issue in a project built outside of nx with `npx create-expo-app my-app`

### Expected Behavior

Able to build an iOS app using `"useFrameworks": "static"`

### Steps to Reproduce

1. Create an empty workspace with either npm or yarn
2. Add an expo app
3. Add a library like React Native Firebase that uses `"useFrameworks": "static"`
4. Prebuild the app  `npm exec -- expo prebuild --clean`
5. Build the iOS app `npm exec -- eas build --profile=development --platform=ios --local`

### Nx Report

```shell
Node : 16.18.1
   OS   : darwin arm64
   npm  : 9.1.3
   
   nx : 15.4.1
   @nrwl/angular : Not Found
   @nrwl/cypress : 15.4.1
   @nrwl/detox : 15.4.1
   @nrwl/devkit : 15.4.1
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.4.1
   @nrwl/expo : 15.4.1
   @nrwl/express : Not Found
   @nrwl/jest : 15.4.1
   @nrwl/js : 15.4.1
   @nrwl/linter : 15.4.1
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 15.4.1
   @nrwl/react-native : Not Found
   @nrwl/rollup : 15.4.1
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : 15.4.1
   @nrwl/webpack : 15.4.1
   @nrwl/workspace : 15.4.1
   @nrwl/vite : 15.4.1
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
```

### app.json
```json
{
  "expo": {
    "name": "Native App",
    "slug": "native-app",
    "version": "1.0.0",
    "jsEngine": "hermes",
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "splash": {
      "image": "./assets/splash.png",
      "resizeMode": "contain",
      "backgroundColor": "#ffffff"
    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": ["**/*"],
    "ios": {
      "supportsTablet": true,
      "bundleIdentifier": "com.native.app",
      "entitlements": {
        "com.apple.developer.devicecheck.appattest-environment": "development",
        "com.apple.developer.networking.wifi-info": true
      },
      "googleServicesFile": "./GoogleService-Info.plist"
    },
    "android": {
      "adaptiveIcon": {
        "foregroundImage": "./assets/adaptive-icon.png",
        "backgroundColor": "#FFFFFF"
      },
      "package": "com.native.app",
      "googleServicesFile": "./google-services.json"
    },
    "web": {
      "favicon": "./assets/favicon.png"
    },
    "extra": {
      "eas": {
        "projectId": ""
      }
    },
    "plugins": [
      "@react-native-firebase/app",
      "@react-native-firebase/perf",
      "@react-native-firebase/crashlytics",
      [
        "expo-build-properties",
        {
          "ios": {
            "useFrameworks": "static"
          }
        }
      ]
    ]
  }
}

```

### Podfile
```
require File.join(File.dirname(`node --print "require.resolve('expo/package.json')"`), "scripts/autolinking")
require File.join(File.dirname(`node --print "require.resolve('react-native/package.json')"`), "scripts/react_native_pods")
require File.join(File.dirname(`node --print "require.resolve('@react-native-community/cli-platform-ios/package.json')"`), "native_modules")

require 'json'
podfile_properties = JSON.parse(File.read(File.join(__dir__, 'Podfile.properties.json'))) rescue {}

platform :ios, podfile_properties['ios.deploymentTarget'] || '13.0'
install! 'cocoapods',
  :deterministic_uuids => false

target 'RnfbApp' do
  use_expo_modules!
  config = use_native_modules!

  use_frameworks! :linkage => podfile_properties['ios.useFrameworks'].to_sym if podfile_properties['ios.useFrameworks']

  # Flags change depending on the env values.
  flags = get_default_flags()

  use_react_native!(
    :path => config[:reactNativePath],
    :hermes_enabled => podfile_properties['expo.jsEngine'] == 'hermes',
    :fabric_enabled => flags[:fabric_enabled],
    # An absolute path to your application root.
    :app_path => "#{Pod::Config.instance.installation_root}/..",
    #
    # Uncomment to opt-in to using Flipper
    # Note that if you have use_frameworks! enabled, Flipper will not work
    # :flipper_configuration => !ENV['CI'] ? FlipperConfiguration.enabled : FlipperConfiguration.disabled,
  )

  post_install do |installer|
    react_native_post_install(
      installer,
      # Set `mac_catalyst_enabled` to `true` in order to apply patches
      # necessary for Mac Catalyst builds
      :mac_catalyst_enabled => false
    )
    __apply_Xcode_12_5_M1_post_install_workaround(installer)

    # This is necessary for Xcode 14, because it signs resource bundles by default
    # when building for devices.
    installer.target_installation_results.pod_target_installation_results
      .each do |pod_name, target_installation_result|
      target_installation_result.resource_bundle_targets.each do |resource_bundle_target|
        resource_bundle_target.build_configurations.each do |config|
          config.build_settings['CODE_SIGNING_ALLOWED'] = 'NO'
        end
      end
    end
  end

  post_integrate do |installer|
    begin
      expo_patch_react_imports!(installer)
    rescue => e
      Pod::UI.warn e
    end
  end
end
```
### Failure Logs

```shell
[RUN_FASTLANE] › Creating  cxxreact.framework
[RUN_FASTLANE] › Copying   react-native ios/build/Build/Products/Debug-iphonesimulator/ReactCommon/ReactCommon.framework/Headers/TurboModuleUtils.h ➜ ../../../../../../var/folders/29/hkk177js0mn_fr5d_bvwmyjr0000gn/T/eas-build-local-nodejs/5c1e7387-965d-4a78-a585-1d4fa87c2907/build/node_modules/react-native/ReactCommon/react/nativemodule/core/ReactCommon/TurboModuleUtils.h
[RUN_FASTLANE] › Copying   react-native ios/build/Build/Products/Debug-iphonesimulator/ReactCommon/ReactCommon.framework/Headers/TurboModulePerfLogger.h ➜ ../../../../../../var/folders/29/hkk177js0mn_fr5d_bvwmyjr0000gn/T/eas-build-local-nodejs/5c1e7387-965d-4a78-a585-1d4fa87c2907/build/node_modules/react-native/ReactCommon/react/nativemodule/core/ReactCommon/TurboModulePerfLogger.h
[RUN_FASTLANE] › Copying   react-native ios/build/Build/Products/Debug-iphonesimulator/ReactCommon/ReactCommon.framework/Headers/TurboModuleBinding.h ➜ ../../../../../../var/folders/29/hkk177js0mn_fr5d_bvwmyjr0000gn/T/eas-build-local-nodejs/5c1e7387-965d-4a78-a585-1d4fa87c2907/build/node_modules/react-native/ReactCommon/react/nativemodule/core/ReactCommon/TurboModuleBinding.h
[RUN_FASTLANE] › Copying   react-native ios/build/Build/Products/Debug-iphonesimulator/ReactCommon/ReactCommon.framework/Headers/TurboModule.h ➜ ../../../../../../var/folders/29/hkk177js0mn_fr5d_bvwmyjr0000gn/T/eas-build-local-nodejs/5c1e7387-965d-4a78-a585-1d4fa87c2907/build/node_modules/react-native/ReactCommon/react/nativemodule/core/ReactCommon/TurboModule.h
[RUN_FASTLANE] › Copying   react-native ios/build/Build/Products/Debug-iphonesimulator/ReactCommon/ReactCommon.framework/Headers/TurboCxxModule.h ➜ ../../../../../../var/folders/29/hkk177js0mn_fr5d_bvwmyjr0000gn/T/eas-build-local-nodejs/5c1e7387-965d-4a78-a585-1d4fa87c2907/build/node_modules/react-native/ReactCommon/react/nativemodule/core/ReactCommon/TurboCxxModule.h
[RUN_FASTLANE] › Copying   react-native ios/build/Build/Products/Debug-iphonesimulator/ReactCommon/ReactCommon.framework/Headers/ReactCommon-umbrella.h ➜ ios/Pods/Target/ Support/ Files/ReactCommon/ReactCommon-umbrella.h
[RUN_FASTLANE] › Copying   react-native ios/build/Build/Products/Debug-iphonesimulator/ReactCommon/ReactCommon.framework/Headers/RCTTurboModuleManager.h ➜ ../../../../../../var/folders/29/hkk177js0mn_fr5d_bvwmyjr0000gn/T/eas-build-local-nodejs/5c1e7387-965d-4a78-a585-1d4fa87c2907/build/node_modules/react-native/ReactCommon/react/nativemodule/core/platform/ios/RCTTurboModuleManager.h
[RUN_FASTLANE] › Copying   react-native ios/build/Build/Products/Debug-iphonesimulator/ReactCommon/ReactCommon.framework/Headers/RCTTurboModule.h ➜ ../../../../../../var/folders/29/hkk177js0mn_fr5d_bvwmyjr0000gn/T/eas-build-local-nodejs/5c1e7387-965d-4a78-a585-1d4fa87c2907/build/node_modules/react-native/ReactCommon/react/nativemodule/core/platform/ios/RCTTurboModule.h
[RUN_FASTLANE] › Copying   react-native ios/build/Build/Products/Debug-iphonesimulator/ReactCommon/ReactCommon.framework/Headers/RCTBlockGuard.h ➜ ../../../../../../var/folders/29/hkk177js0mn_fr5d_bvwmyjr0000gn/T/eas-build-local-nodejs/5c1e7387-965d-4a78-a585-1d4fa87c2907/build/node_modules/react-native/ReactCommon/react/nativemodule/core/platform/ios/RCTBlockGuard.h
[RUN_FASTLANE] › Copying   react-native ios/build/Build/Products/Debug-iphonesimulator/ReactCommon/ReactCommon.framework/Headers/LongLivedObject.h ➜ ../../../../../../var/folders/29/hkk177js0mn_fr5d_bvwmyjr0000gn/T/eas-build-local-nodejs/5c1e7387-965d-4a78-a585-1d4fa87c2907/build/node_modules/react-native/ReactCommon/react/nativemodule/core/ReactCommon/LongLivedObject.h
[RUN_FASTLANE] › Compiling react-native Pods/ReactCommon » TurboModuleUtils.cpp
[RUN_FASTLANE] 
❌  (../../node_modules/react-native/ReactCommon/react/nativemodule/core/ReactCommon/TurboModuleUtils.h:16:10)

  14 | 
  15 | #include <ReactCommon/CallInvoker.h>
> 16 | #include <react/bridging/CallbackWrapper.h>
     |          ^ 'react/bridging/CallbackWrapper.h' file not found
  17 | #include <react/bridging/LongLivedObject.h>
  18 | 
  19 | namespace facebook {

[RUN_FASTLANE] › Compiling react-native Pods/ReactCommon » TurboModulePerfLogger.cpp
[RUN_FASTLANE] › Compiling react-native Pods/ReactCommon » TurboModuleBinding.cpp
[RUN_FASTLANE] 
❌  (../../node_modules/react-native/ReactCommon/react/nativemodule/core/ReactCommon/LongLivedObject.h:11:10)

   9 | 
  10 | // This header is left here for compatibility reasons.
> 11 | #include <react/bridging/LongLivedObject.h>
     |          ^ 'react/bridging/LongLivedObject.h' file not found
  12 | 

[RUN_FASTLANE] › Compiling react-native Pods/ReactCommon » TurboModule.cpp
[RUN_FASTLANE] › Compiling react-native Pods/ReactCommon » TurboCxxModule.cpp
[RUN_FASTLANE] › Compiling react-native Pods/ReactCommon » ReactCommon_vers.c
[RUN_FASTLANE] ▸ ** BUILD FAILED **
[RUN_FASTLANE] ▸ The following build commands failed:
[RUN_FASTLANE] ▸        CompileC /var/folders/29/hkk177js0mn_fr5d_bvwmyjr0000gn/T/eas-build-local-nodejs/5c1e7387-965d-4a78-a585-1d4fa87c2907/build/apps/native-app/ios/build/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/ReactCommon.build/Objects-normal/x86_64/TurboModuleUtils.o /var/folders/29/hkk177js0mn_fr5d_bvwmyjr0000gn/T/eas-build-local-nodejs/5c1e7387-965d-4a78-a585-1d4fa87c2907/build/node_modules/react-native/ReactCommon/react/nativemodule/core/ReactCommon/TurboModuleUtils.cpp normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'ReactCommon' from project 'Pods')
[RUN_FASTLANE] ▸ (1 failure)
[RUN_FASTLANE] › Compiling react-native Pods/ReactCommon » ReactCommon-dummy.m
[RUN_FASTLANE] ** BUILD FAILED **
[RUN_FASTLANE] 
[RUN_FASTLANE] 
[RUN_FASTLANE] The following build commands failed:
[RUN_FASTLANE]  CompileC /var/folders/29/hkk177js0mn_fr5d_bvwmyjr0000gn/T/eas-build-local-nodejs/5c1e7387-965d-4a78-a585-1d4fa87c2907/build/apps/native-app/ios/build/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/ReactCommon.build/Objects-normal/x86_64/TurboModuleUtils.o /var/folders/29/hkk177js0mn_fr5d_bvwmyjr0000gn/T/eas-build-local-nodejs/5c1e7387-965d-4a78-a585-1d4fa87c2907/build/node_modules/react-native/ReactCommon/react/nativemodule/core/ReactCommon/TurboModuleUtils.cpp normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'ReactCommon' from project 'Pods')
[RUN_FASTLANE] (1 failure)
[RUN_FASTLANE] Exit status: 65
```

**Comment by atanaskanchev** [comment]
Still no luck.. I wonder if the way we handle the node_modules folder in the scripts` "postinstall": "rm -r node_modules && cd ../../ && yarn install && npx nx sync-deps native-app && npx nx ensure-symlink native-app",` could be a possible culprit. Symlinking the root node_modules is the only major difference with a naked expo project

**Comment by atanaskanchev** [comment]
Hi @AgentEnder  [reproducible example]( https://github.com/atanaskanchev/expo-nx-static-frameworks). This issue still persist

```
    System:
      OS: macOS 13.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.18.1 - ~/.nvm/versions/node/v16.18.1/bin/node
      Yarn: 1.22.19 - ~/.nvm/versions/node/v16.18.1/bin/yarn
      npm: 9.1.3 - ~/.nvm/versions/node/v16.18.1/bin/npm
    Managers:
      CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    IDEs:
      Android Studio: 2021.3 AI-213.7172.25.2113.9123335
      Xcode: 14.2/14C18 - /usr/bin/xcodebuild
    npmPackages:
      @expo/metro-config: * => 0.5.2 
      @expo/webpack-config: * => 0.17.3 
      expo: * => 47.0.10 
      react: * => 18.2.0 
      react-native: * => 0.70.6 
      react-native-web: * => 0.18.10
      ```

**Comment by atanaskanchev** [comment]
Still an issue using `15.4.4`

```shell
[RUN_FASTLANE] › Compiling react-native Pods/ReactCommon » TurboModuleUtils.cpp
[RUN_FASTLANE] 
❌  (../../node_modules/react-native/ReactCommon/react/nativemodule/core/ReactCommon/TurboModuleUtils.h:16:10)

  14 | 
  15 | #include <ReactCommon/CallInvoker.h>
> 16 | #include <react/bridging/CallbackWrapper.h>
     |          ^ 'react/bridging/CallbackWrapper.h' file not found
  17 | #include <react/bridging/LongLivedObject.h>
  18 | 
  19 | namespace facebook {

[RUN_FASTLANE] › Compiling react-native Pods/ReactCommon » TurboModulePerfLogger.cpp
[RUN_FASTLANE] › Compiling react-native Pods/ReactCommon » TurboModule.cpp
[RUN_FASTLANE] › Compiling react-native Pods/ReactCommon » TurboModuleBinding.cpp
[RUN_FASTLANE] 
❌  (../../node_modules/react-native/ReactCommon/react/nativemodule/core/ReactCommon/LongLivedObject.h:11:10)

   9 | 
  10 | // This header is left here for compatibility reasons.
> 11 | #include <react/bridging/LongLivedObject.h>
     |          ^ 'react/bridging/LongLivedObject.h' file not found
  12 | 

[RUN_FASTLANE] › Compiling react-native Pods/ReactCommon » TurboCxxModule.cpp
[RUN_FASTLANE] › Compiling react-native Pods/ReactCommon » ReactCommon-dummy.m
[RUN_FASTLANE] › Compiling react-native Pods/ReactCommon » RCTTurboModuleManager.mm
[RUN_FASTLANE] › Compiling react-native Pods/ReactCommon » RCTBlockGuard.mm
[RUN_FASTLANE] ▸ ** ARCHIVE FAILED **
[RUN_FASTLANE] ▸ The following build commands failed:
[RUN_FASTLANE] ▸        CompileC /Users/atanaskanchev/Library/Developer/Xcode/DerivedData/NativeApp-hjnbyejqemvfdzgqranuepmcszje/Build/Intermediates.noindex/ArchiveIntermediates/NativeApp/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/ReactCommon.build/Objects-normal/arm64/TurboModuleUtils.o /var/folders/29/hkk177js0mn_fr5d_bvwmyjr0000gn/T/eas-build-local-nodejs/386d0740-0c84-4f8f-b4c1-0489df52f245/build/node_modules/react-native/ReactCommon/react/nativemodule/core/ReactCommon/TurboModuleUtils.cpp normal arm64 c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'ReactCommon' from project 'Pods')
[RUN_FASTLANE] ▸ (1 failure)
[RUN_FASTLANE] › Compiling react-native Pods/ReactCommon » RCTTurboModule.mm
[RUN_FASTLANE] › Preparing react-native Pods/RCTTypeSafety » RCTTypeSafety-Info.plist
[RUN_FASTLANE] › Compiling react-native Pods/ReactCommon » ReactCommon_vers.c
[RUN_FASTLANE] ** ARCHIVE FAILED **
[RUN_FASTLANE] 
[RUN_FASTLANE] 
[RUN_FASTLANE] The following build commands failed:
[RUN_FASTLANE]  CompileC /Users/atanaskanchev/Library/Developer/Xcode/DerivedData/NativeApp-hjnbyejqemvfdzgqranuepmcszje/Build/Intermediates.noindex/ArchiveIntermediates/NativeApp/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/ReactCommon.build/Objects-normal/arm64/TurboModuleUtils.o /var/folders/29/hkk177js0mn_fr5d_bvwmyjr0000gn/T/eas-build-local-nodejs/386d0740-0c84-4f8f-b4c1-0489df52f245/build/node_modules/react-native/ReactCommon/react/nativemodule/core/ReactCommon/TurboModuleUtils.cpp normal arm64 c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'ReactCommon' from project 'Pods')
```

```
   Node : 16.18.1
   OS   : darwin arm64
   yarn : 1.22.19
   
   nx : 15.4.4
   @nrwl/angular : Not Found
   @nrwl/cypress : 15.4.4
   @nrwl/detox : 15.4.4
   @nrwl/devkit : 15.4.4
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.4.4
   @nrwl/expo : 15.4.4
   @nrwl/express : Not Found
   @nrwl/jest : 15.4.4
   @nrwl/js : 15.4.4
   @nrwl/linter : 15.4.4
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 15.4.4
   @nrwl/react-native : Not Found
   @nrwl/rollup : 15.4.4
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : 15.4.4
   @nrwl/webpack : 15.4.4
   @nrwl/workspace : 15.4.4
   @nrwl/vite : 15.4.4
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
```

**Comment by ste7en** [comment]
even without static linkage doesnt'work (v15.4.2)

**Comment by ste7en** [comment]
As a temporary fix I was able to build the project with the manual fix pointed by @evelant in the Manual Fix subsection of these issues: https://github.com/pnpm/pnpm/issues/5385, https://github.com/invertase/react-native-firebase/issues/6561, https://github.com/expo/expo/issues/19200, https://github.com/facebook/react-native/issues/34742.

> **Manual fix**
> 
> If you change the copy phase of the `React-bridging` pod to an absolute path of `$(PODS_ROOT)/Headers/Private/React-bridging/react/bridging` the headers get copied to the correct location. This fixes building of the pods project but the main project still fails.
> 
> Add `$(PODS_ROOT)/Headers/Private/React-bridging` to the main project header search paths and the project will finally build.

**Comment by evelant** [comment]
This is fixed in react-native 0.71. Since we can't use that with expo yet I created a patch that backports the fixes so that pnpm users can compile with `use_frameworks!`.

Patch:
[react-native@0.70.6.patch](https://github.com/nrwl/nx/files/10366467/react-native%400.70.6.patch)

**Comment by ste7en** [comment]
thanks @evelant but why are you referencing the React-jsc podspec in your patch if it does not exist for react-native ≤0.71 ?
Indeed when I `pod install` I get 
```
[!] No podspec found for `React-jsc` in `../node_modules/react-native/ReactCommon/jsc`
```

**Comment by aureliopetrone** [comment]
@evelant what about Yarn users? Can I use the patch with Yarn?

**Comment by RohanWeli** [comment]
I also have the same problem. I'm also using Nx with Yarn along with Expo and Firebase(with useFrameworks).

@ste7en appreciate if you can provide more details on how to do the manual fix. 

specially on how to change the copy phase of the React-bridging pod to an absolute path of $(PODS_ROOT)/Headers/Private/React-bridging/react/bridging the headers get copied to the correct location.

**Comment by ste7en** [comment]
@RohanWeli you can use this patch on react-native 0.70.6: 
[react-native@0.70.6.patch](https://github.com/nrwl/nx/files/10555283/react-native%400.70.6.patch)

Just run `patch -p1 < react-native@0.70.6.patch` inside your `node_modules/react-native` directory (or temporary directory) after you've started the patching process through your package manager (I guess you're going to use patch-package for Yarn 1, yarn patch for Yarn ≥2).

Otherwise, if you're following the manual fix, choose `React-bridging` target from your Pods project, go to `Build Phases` tab and, from the `Copy xxx...` phase, select `Absolute Path` as Destination. The, add $(PODS_ROOT)/Headers/Private/React-bridging to the main project header search paths and the project will finally build. But the next time you're going to `pod install` your project this must be re-applied.

**Comment by aureliopetrone** [comment]
Ciao Stefano @ste7en,

after changing the headers path to absolute and adding the headers like this:

<img width="878" alt="Screenshot 2023-02-01 at 10 56 06" src="https://user-images.githubusercontent.com/12773128/216010604-960eaf7a-e39a-4460-8e6b-715d093c7619.png">

I get this errors now:

<img width="490" alt="Screenshot 2023-02-01 at 10 53 18" src="https://user-images.githubusercontent.com/12773128/216009973-fbd4629a-33fa-4712-af01-430c3261a790.png">

Thanks for your support, this is blocking us since a week.

**Comment by ste7en** [comment]
Did you try to clean the build folder and rebuild?

**Comment by aureliopetrone** [comment]
Yes, same thing.

<img width="485" alt="Screenshot 2023-02-01 at 11 06 47" src="https://user-images.githubusercontent.com/12773128/216013082-aac81f46-ebf9-46d5-aa34-9467f62e62d8.png">

**Comment by ste7en** [comment]
Change the absolute path destination to `$(PODS_ROOT)/Headers/Private/React-bridging/react/bridging`

**Comment by aureliopetrone** [comment]
After doing that, cleaning and rebuild, I still get this.

<img width="477" alt="Screenshot 2023-02-01 at 11 20 09" src="https://user-images.githubusercontent.com/12773128/216016185-78d32cbb-f0a8-4e32-8a5c-7cfceea3300a.png">

**Comment by ste7en** [comment]
If you also cleaned and rebuilt your Pods project you need to reapply the fix. Otherwise don't know, which version of rn are you using?

**Comment by aureliopetrone** [comment]
@ste7en  Does the structure of this folder could be the problem?

<img width="446" alt="Screenshot 2023-02-01 at 11 24 19" src="https://user-images.githubusercontent.com/12773128/216017143-a38050dc-4e86-46e8-9e1e-87547cf2d394.png">

**Comment by ste7en** [comment]
It shouldn't be a problem, I guess it is due to Nx. Are you applying Nx post_install hook in your Podfile?
Try also with `nx sync-deps` before installing your pods, then clean, apply the fix and rebuild

**Comment by aureliopetrone** [comment]
@ste7en Yes, tried also to install pod without nx_post_install and building either from nx script and Xcode, same. I'm going crazy with this issue.

**Comment by ste7en** [comment]
@aureliopetrone Did you enable Hermes?
I know it's so annoying... apply the patch if it keeps showing that error

**Comment by aureliopetrone** [comment]
@ste7en I added you on LinkedIn, if you like my company has allowed me to hire someone to solve this problem.

**Comment by RohanWeli** [comment]
Thanks @ste7en for the support. I have managed to get the iOS build working manually via Xcode with the manual fix.

when hermese is enabled, I had to 
`
use_frameworks! :linkage => :static
`

when hermese is disabled
`use_frameworks!
`
in both case 
`
    __apply_Xcode_12_5_M1_post_install_workaround(installer)

    installer.pods_project.targets.each do |target|
      target.build_configurations.each do |config|
        config.build_settings["HEADER_SEARCH_PATHS"] ||= "$(inherited) "
        config.build_settings["HEADER_SEARCH_PATHS"] << '"$(PODS_TARGET_SRCROOT)" '
        config.build_settings['HEADER_SEARCH_PATHS'] << '"$(PODS_ROOT)/Headers/Private/React-bridging" '
      end
    end

    nx_post_install(installer)
`
my package versions are
`
"dependencies": {
    "@react-native-firebase/app": "^16.7.0",
    "@react-native-firebase/crashlytics": "^16.7.0",
    "expo": "^47.0.13",
    "expo-constants": "^14.0.2",
    "patch-package": "^6.5.1",
    "postinstall-postinstall": "^2.1.0",
    "react": "18.1.0",
    "react-dom": "18.1.0",
    "react-native": "0.70.6",
    "tslib": "^2.3.0"
  }

`

**Comment by kdenz** [comment]
I'm using `pnpm` + RN 0.70.6
Fixed as per @evelant 's patch  https://github.com/nrwl/nx/issues/14014#issuecomment-1374497393 (Thanks man!)

**Comment by dedan** [comment]
Hello @evelant, thanks a lot for your patch and support on this thread. I applied your patch, but after that receive the following error:

```
⚠️  Something went wrong running `pod install` in the `ios` directory.
Command `pod install` failed.
└─ Cause: No podspec found for `React-jsc` in `../node_modules/react-native/ReactCommon/jsc`
```

Looking at line 1756 of your patch, I see that the file `ReactCommon/React-bridging.podspec` should have been copied and renamed to `ReactCommon/jsc/React-jsc.podspec`. However, this did not happen when I applied your patch with `patch -p1 < react-native@0.70.6.patch` within the `react-native` directory. 

Is there anything else I should have done? Sorry if it's an obvious mistake, I'm not familiar with applying patches by hand. I made sure that I'm using `react-native@0.70.6`. 

Also, here is the output from running the patch command: https://app.warp.dev/block/yxcG8JpTaMVayLJmfKQ84V
